opengever/setup/sections/fileloader.py

```
@@ -0,0 +1,109 @@
+
+        # TODO: Handle .eml / ftw.mail.mail
+        self.file_field_class = IDocumentSchema['file']._type
+
+        self.stats = IAnnotations(transmogrifier)[JSON_STATS_KEY]
+        self.stats['errors']['files_not_found'] = {}
+        self.stats['errors']['files_io_errors'] = {}
+        self.stats['msgs'] = {}
+
+    def __iter__(self):
+        for item in self.previous:
+            keys = item.keys()
+            pathkey = self.pathkey(*keys)[0]
+
+            if self.key in item:
+                filepath = item[self.key]
+                if filepath is None:
+                    yield item
+                    continue
+
+                if pathkey not in item:
+                    logger.warning("Missing path key for file %s" % filepath)
+                    yield item
+                    continue
+                path = item[pathkey]
+
+                # TODO: Check for this in OGGBundle validation
+                if filepath.endswith(u'.msg'):
+                    logger.warning("Skipping .msg file: %s" % filepath)
+                    self.stats['msgs'][path] = filepath
+                    yield item
+                    continue
+
+                filepath = os.path.join(self.bundle_path, filepath)
+                filename = os.path.basename(filepath)
+
+                # TODO: Check for this in OGGBundle validation
+                if not os.path.exists(filepath):
+                    logger.warning("File not found: %s" % filepath)
+                    self.stats['errors']['files_not_found'][filepath] = path
+                    yield item
+                    continue
+
+                mimetype, _encoding = guess_type(filepath, strict=False)
+                if mimetype is None:
+                    logger.warning("Unknown mimetype for file %s" % filepath)
+                    mimetype = 'application/octet-stream'
+
+                obj = traverse(self.context, path, None)
+                if obj is None:
+                    logger.warning(
```

this should not happen, ever. maybe also fail hard?